plugins/shrs_completion/src/lib.rs

```diff
@@ -4,3 +4,65 @@
 
 pub mod completions;
 mod helpers;
+use std::{fs, rc::Rc};
+
+use ::anyhow::anyhow;
+use dirs::home_dir;
+use rhai::{Array, Engine, Scope};
+use setup::setup_engine;
+use shrs::prelude::*;
+
+pub struct CompletionsPlugin;
+pub mod setup;
+
+impl Plugin for CompletionsPlugin {
+    fn init(&self, _: &mut ShellConfig) -> anyhow::Result<()> {
+        Ok(())
+    }
+    fn post_init(&self, sh: &Shell, ctx: &mut Context, rt: &mut Runtime) -> ::anyhow::Result<()> {
+        let mut e = Engine::new();
+        setup_engine(&mut e);
+        let engine = Rc::new(e);
+        let mut folder = home_dir().unwrap();
+        folder.push(".config/shrs/completions");
```

Comment:
should create this directory if it doesn't exist; abstract out this dir setup logic to a helper